torch/csrc/utils/python_arg_parser.h

```diff
@@ -813,8 +813,14 @@ inline at::Device toDevice(PyObject* obj) {
           c10::DeviceType::PrivateUse1,
           static_cast<c10::DeviceIndex>(device_index));
     }
+#ifdef USE_CUDA
```

Comment:
Updated https://github.com/pytorch/pytorch/issues/126646 with details.
Can you change this to get the current accelerator instead?

Reply:
OK, using the current accelerator looks better.
And I add `XPU` to `getAccelerator` and refine the code.